### DISCUSSION TITLE
VSCode: Unable to use Remote-SSH with Lima VM after updating macOS to 13.4

### DISCUSSION THREAD

**Discussion by parlarlax** [lima-vm/lima#1605] [Q&A] [answered]
## Description
After updating my macOS to version 13.4, I am experiencing issues with code-server (vscode-remote-ssh) when trying to connect to a Lima VM. However, I was able to successfully use the same instance of code-server before the macOS update. I can still SSH into the Lima VM using the provided configuration without any problems, but the issue arises specifically when attempting to connect via code-server.

## Steps to reproduce
1. Update macOS to version 13.4.
2. Attempt to connect to the Lima VM using Remote-SSH.
3. Observe the issue/error that occurs.

## Expected behavior
I expect to be able to connect to the Lima VM using remote-ssh as I did before the macOS update.

## Actual behavior
Since updating to macOS 13.4, I am unable to establish a connection to the Lima VM using code-server (vscode-remote-ssh). The connection process fails, and I am unable to access the remote development environment. However, SSH connections to the Lima VM using the provided configuration still work fine.

### SSH Configuration
In my SSH configuration file, I have the following entry for connecting to the Lima VM:

```
Host lima-docker
    HostName 127.0.0.1
    Port 60022
    IdentityFile ~/.lima/_config/user
    NoHostAuthenticationForLocalhost yes
```

### Successful SSH Command
I can successfully connect to the Lima VM using the following SSH command:

`ssh -p 60022 -i ~/.lima/_config/user -o NoHostAuthenticationForLocalhost=yes 127.0.0.1`

## Error messages and logs
When starting code-server manually on vm, I receive the following error:
```
*
* Visual Studio Code Server
*
* By using the software, you agree to
* the Visual Studio Code Server License Terms (https://aka.ms/vscode-server-license) and
* the Microsoft Privacy Statement (https://privacy.microsoft.com/en-US/privacystatement).
*
Do you accept the terms in the License Agreement (Y/n)?
Server bound to 127.0.0.1:8000 (IPv4)
Extension host agent listening on 8000

terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)
```

When starting code-server using the SSH remote extension in Visual Studio Code, I encounter the following error message within the editor:

```
Failed to set up socket for dynamic port forward to remote port 33833: connect ECONNREFUSED 127.0.0.1:58699. Is the remote port correct?
```

## Attempts and research
I have tried restarting my Mac and the Lima VM, but the issue persists. I have not made any changes to my Lima VM configuration or code-server setup. I have also researched online for any similar issues or known compatibility problems between code-server, Lima VM, and macOS 13.4, but could not find any relevant information.

## Host Environment
- macOS version: 13.4
- CPU ARCH: aarm64 [M1]
- Lima version: 0.16.0
- VSCode version: 1.78

## Lima Instance Environment
- Ubuntu version: 22.04
- CPU ARCH: x86_64
- Code-Server version: 4.13.0

## Additional information
- SSH connections to the Lima VM using the provided configuration are working fine.
- The problem only occurs when attempting to use code-server (vscode-remote-ssh).
- The error logs suggest a potential issue with socket setup and dynamic port forwarding.
- Considering downgrading to a previous macOS version as a possible solution.

Any suggestions or insights on how to troubleshoot or resolve this issue would be greatly appreciated.

**Comment by balajiv113** [comment]
Hi, could you also share your lima configuration that you are using ?

**Reply by balajiv113** [reply, marked as answer]
From the above info i guess you are using emulated qemu lima instance on M1.

There is already a similar open issue with this combination 
https://github.com/microsoft/vscode-remote-release/issues/8391

Maybe one suggestion would be try with vz driver with rosetta enabled. See if that helps

**Reply by parlarlax** [reply]
Hi there! Absolutely, I'd be happy to share my Lima configuration with you. I use the following command to start my Lima instance

```limactl start --name=default template://docker --set='.cpus = 4 | .memory = "4GiB" | .arch = "x86_64"' ```

Regarding your suggestion, I will definitely give it a try. I'll attempt to use the vz driver with Rosetta enabled and see if that helps resolve the issue. Thank you for the suggestion
